packages/ingest/src/granule.ts

```diff
@@ -1,71 +1,109 @@
+import moment from 'moment';
+import * as S3 from '@cumulus/aws-client/S3';
+import * as log from '@cumulus/common/log';
+import { DuplicateHandling } from '@cumulus/types';
+
+export interface EventWithDuplicateHandling {
+  config: {
+    collection: {
+      duplicateHandling?: DuplicateHandling
+    },
+    duplicateHandling?: DuplicateHandling,
+  },
+  cumulus_config?: {
+    cumulus_context?: {
+      forceDuplicateOverwrite?: boolean
+    }
+  },
+}
 
-const encodeurl = require('encodeurl');
-const errors = require('@cumulus/errors');
-const get = require('lodash/get');
-const moment = require('moment');
-const S3 = require('@cumulus/aws-client/S3');
-const log = require('@cumulus/common/log');
-const { deprecate } = require('@cumulus/common/util');
+export interface File {
+  bucket?: string,
+  key?: string,
+  fileName?: string,
+  name?: string,
+  filename?: string
+}
+
+export interface MoveFileParams {
+  source?: {
+    Bucket: string,
+    Key: string
+  },
+  target?: {
+    Bucket: string,
+    Key: string
+  },
+  file: File
+}
 
 /**
-* Copy granule file from one s3 bucket & keypath to another
-*
-* @param {Object} source - source
-* @param {string} source.Bucket - source
-* @param {string} source.Key - source
-* @param {Object} target - target
-* @param {string} target.Bucket - target
-* @param {string} target.Key - target
-* @param {Object} options - optional object with properties as defined by AWS API:
-* https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#copyObject-property
-* @returns {Promise} returms a promise that is resolved when the file is copied
-**/
-function copyGranuleFile(source, target, options = {}) {
-  deprecate(
-    '@cumulus/ingest/granule.copyGranuleFile',
-    '1.22.1',
-    '@cumulus/aws-client/S3.multipartCopyObject'
-  );
+  * rename s3 file with timestamp
+  *
+  * @param {string} bucket - bucket of the file
+  * @param {string} key - s3 key of the file
+  * @returns {Promise} promise that resolves when file is renamed
+  */
+export async function renameS3FileWithTimestamp(
+  bucket: string,
+  key: string
+): Promise<void> {
+  const formatString = 'YYYYMMDDTHHmmssSSS';
+  const timestamp = (await S3.headObject(bucket, key)).LastModified;
 
-  const CopySource = encodeurl(`${source.Bucket}/${source.Key}`);
+  if (!timestamp) {
+    throw new Error(`s3://${bucket}/${key} does not have a LastModified property`);
+  }
 
-  const params = {
-    CopySource,
-    Bucket: target.Bucket,
-    Key: target.Key,
-    ...options
-  };
+  let renamedKey = `${key}.v${moment.utc(timestamp).format(formatString)}`;
 
-  return S3.s3CopyObject(params)
-    .catch((error) => {
-      log.error(`Failed to copy s3://${CopySource} to s3://${target.Bucket}/${target.Key}: ${error.message}`);
-      throw error;
-    });
+  // if the renamed file already exists, get a new name
+  // eslint-disable-next-line no-await-in-loop
+  while (await S3.s3ObjectExists({ Bucket: bucket, Key: renamedKey })) {
```

Comment:
Why is this a `while` insted of a basic `if`?

Reply:
It looks like this will handle cases where there are already multiple versions of a file. It will continue increasing the number until it finds one that doesn't exist yet. I agree that it should be fixed, but that refactor felt beyond the scope of this PR.